src/main.rs

```diff
@@ -146,6 +159,7 @@ async fn event_loop(
         if !keys.sent_info {
             let content: String = METHODS
                 .iter()
+                .filter(|i| !config.recv_only() || RECV_ONLY_METHODS.contains(i))
```

Comment:
there's probably a smarter way than to filter `METHODS` at multiple places and check for `config.recv_only()`.

Reply:
suggested a cleaner way